vpr/src/route/rr_graph2.cpp

```diff
@@ -1203,53 +1203,53 @@ bool verify_rr_node_indices(const DeviceGrid& grid,
                     }
 
                     if (rr_graph.node_type(inode) == CHANX) {
-                        VTR_ASSERT_MSG(rr_node.ylow() == rr_node.yhigh(), "CHANX should be horizontal");
+                        VTR_ASSERT_MSG(rr_graph.node_ylow(rr_node.id()) == rr_graph.node_yhigh(rr_node.id()), "CHANX should be horizontal");
 
-                        if (y != rr_node.ylow()) {
+                        if (y != rr_graph.node_ylow(rr_node.id())) {
                             VPR_ERROR(VPR_ERROR_ROUTE, "RR node y position does not agree between rr_nodes (%d) and rr_node_indices (%d): %s",
-                                      rr_node.ylow(),
+                                      rr_graph.node_ylow(rr_node.id()),
                                       y,
                                       describe_rr_node(size_t(inode)).c_str());
                         }
 
-                        if (x < rr_node.xlow() || x > rr_node.xhigh()) {
+                        if (x < rr_graph.node_xlow(rr_node.id()) || x > rr_graph.node_xhigh(rr_node.id())) {
                             VPR_ERROR(VPR_ERROR_ROUTE, "RR node x positions do not agree between rr_nodes (%d <-> %d) and rr_node_indices (%d): %s",
-                                      rr_node.xlow(),
-                                      rr_node.xlow(),
+                                      rr_graph.node_xlow(rr_node.id()),
+                                      rr_graph.node_xlow(rr_node.id()),
                                       x,
                                       describe_rr_node(size_t(inode)).c_str());
                         }
                     } else if (rr_graph.node_type(inode) == CHANY) {
-                        VTR_ASSERT_MSG(rr_node.xlow() == rr_node.xhigh(), "CHANY should be veritcal");
+                        VTR_ASSERT_MSG(rr_graph.node_xlow(rr_node.id()) == rr_graph.node_xhigh(rr_node.id()), "CHANY should be veritcal");
 
-                        if (x != rr_node.xlow()) {
+                        if (x != rr_graph.node_xlow(rr_node.id())) {
                             VPR_ERROR(VPR_ERROR_ROUTE, "RR node x position does not agree between rr_nodes (%d) and rr_node_indices (%d): %s",
-                                      rr_node.xlow(),
+                                      rr_graph.node_xlow(rr_node.id()),
                                       x,
                                       describe_rr_node(size_t(inode)).c_str());
                         }
 
-                        if (y < rr_node.ylow() || y > rr_node.yhigh()) {
+                        if (y < rr_graph.node_ylow(rr_node.id()) || y > rr_graph.node_yhigh(rr_node.id())) {
```

Comment:
This could be an API which judges if a given ``y`` is in range, since it is frequently used.